### PR TITLE
Add slack notifications

### PR DESCRIPTION
**Pull request by senechal**

### CODE REVIEW

File: raccoon_notifier/email_sender.py [hunk 1]
```diff
@@ -14,48 +14,52 @@ def __init__(self, name, user, pwd):
...
             logging.error(e)
+            raise
         self.server.ehlo()
         self.server.starttls()
         body_type = 'html' if force_html else 'plain'
         try:
             self.server.login(self.gmail_user, self.gmail_pwd)
         except smtplib.SMTPAuthenticationError:
             logging.error('Authentication Error: email or password incorrect')
+            raise
```
Comment (arielmorelli) [inline raccoon_notifier/email_sender.py]: Shouldn't raise a more generic exception?
Without the generic exception the developer must import *smtplib* or *smtplib.SMTPAuthenticationError* to treat this case.

File: raccoon_notifier/email_sender.py [hunk 2]
```diff
@@ -14,48 +14,52 @@ def __init__(self, name, user, pwd):
...
-            msg.attach(header)
+        header = MIMEText(self.header, 'html')
+        msg.attach(header)
 
         try:
             msg1 = MIMEText(body, body_type, 'utf-8')
             msg.attach(msg1)
         except UnicodeEncodeError:
             logging.error('Unicode error sending email')
+            raise
```
Comment (arielmorelli) [inline raccoon_notifier/email_sender.py]: Maybe generic exception here too?

File: raccoon_notifier/email_sender.py [hunk 3]
```diff
@@ -66,16 +70,18 @@ def send(self, toaddrs, subject, body, footer=True, force_html=False, attachment
                     msg.attach(img)
             except IOError as e:
                 logging.error("Couldn't inline image. " + str(e))
+                raise
```
Comment (arielmorelli) [inline raccoon_notifier/email_sender.py]: Generic exception here too.

File: raccoon_notifier/email_sender.py [hunk 4]
```diff
@@ -66,16 +70,18 @@ def send(self, toaddrs, subject, body, footer=True, force_html=False, attachment
...
             logging.debug('DEBUG MODE: Email: {0}'.format(msg))
         else:
             try:
-                response = self.server.sendmail(msg['From'], toaddrs, msg.as_string())
+                result = self.server.sendmail(msg['From'], toaddrs, msg.as_string())
                 logging.info('Sent email to: {0}.'.format(toaddrs))
             except smtplib.SMTPException:
                 logging.error('Unable to send email to %s' % toaddrs)
-
+                raise
```
Comment (arielmorelli) [inline raccoon_notifier/email_sender.py]: Generic exception here too.

File: raccoon_notifier/slack_sender.py [hunk 5]
```diff
@@ -0,0 +1,28 @@
...
+                "text": message,
+                "channel": channel,
+            }
+            result = requests.post(self.hook, json=params, headers=self.header)
+            if result.status_code == 400:
+                logging.error("Unable to send Slack Message.")
+                result.raise_for_status()
+                return result
+            return result
+            logging.info("Sent Slack message to {}.".format(channel))
```
Comment (arielmorelli) [inline raccoon_notifier/slack_sender.py]: Swapped lines 27 and 28

File: raccoon_notifier/sms_sender.py [hunk 6]
```diff
@@ -1,4 +1,4 @@
-"Define class SMSSender, used to send sms messages."
+"""Define class SMSSender, used to send sms messages."""
```
Comment (arielmorelli) [inline raccoon_notifier/sms_sender.py]: Blame on me this one. :disappointed:

File: raccoon_notifier/sms_sender.py [hunk 7]
```diff
@@ -40,8 +40,9 @@ def send(self, phones, message):
 
             ids = [reply["id"] for reply in replies["messages"]]
 
-            if result.status_code == 400:
+            if result.status_code != 200:
```
Comment (arielmorelli) [inline raccoon_notifier/sms_sender.py]: Shouldn't be 2xx code?

```200 <= result.status_code < 300```

File: requirements.txt [hunk 8]
```diff
@@ -1,3 +1,5 @@
 requests
 urllib3[secure]
 pyOpenSSL
+nose
+minimock
```
Comment (arielmorelli) [inline requirements.txt]: Missing the current version of the requirements. (yeah, it's a old "must be fixed next update" :see_no_evil: )
```
pip freeze > requirements.txt
```
and removing ```pkg-resources==0.0.0```

does the trick  :grimacing:

File: tests/test_email_sender.py [hunk 9]
```diff
@@ -0,0 +1,88 @@
...
+        self.username = 'test'
+        self.password = '123456'
+        self.toaddrs = ['test@test.com']
+        self.subject = 'Test'
+        self.body = 'Hakuna Matata'
+
+    @patch('smtplib.SMTP')
+    def test_send_without_attachment(self, mock_smtp):
+        mock_sendmail = MagicMock()
+        mock_sendmail.return_value = {}
```
Comment (arielmorelli) [inline tests/test_email_sender.py]: Just telling: You can use the @*patch* to *return_value*, *side_effect* and etc, but here works fine too!

File: raccoon_notifier/email_sender.py (hunk 1 above)
Comment (senechal) [inline raccoon_notifier/email_sender.py]: I don't Think so, the developer can catch as a generic exception on his part, i think is better to catch the expected exception here.

File: requirements.txt (hunk 8 above)
Comment (senechal) [inline requirements.txt]: ok

Comment (senechal) [inline requirements.txt]: I know how pip freeze works ¬¬'

File: raccoon_notifier/slack_sender.py (hunk 5 above)
Comment (senechal) [inline raccoon_notifier/slack_sender.py]: yeah, i didnt see that, thanks.

File: raccoon_notifier/sms_sender.py (hunk 7 above)
Comment (senechal) [inline raccoon_notifier/sms_sender.py]: if correct is 20xx, this would be result.status_code >= 300, no?

Comment (senechal) [inline raccoon_notifier/sms_sender.py]: besides, i think in the context only 200 apply.

File: raccoon_notifier/email_sender.py (hunk 1 above)
Comment (arielmorelli) [inline raccoon_notifier/email_sender.py]: Makes sense.

File: raccoon_notifier/sms_sender.py (hunk 7 above)
Comment (arielmorelli) [inline raccoon_notifier/sms_sender.py]: Makes sense too, I was not certain when i comment this.

**Pull request merged**